src/osd/PrimaryLogPG.cc

```diff
@@ -9976,6 +9978,264 @@ struct C_Flush : public Context {
   }
 };
 
+struct C_SetDedupChunks : public Context {
+  PrimaryLogPGRef pg;
+  hobject_t oid;
+  epoch_t last_peering_reset;
+  ceph_tid_t tid;
+  uint64_t offset;
+  
+  C_SetDedupChunks(PrimaryLogPG *p, hobject_t o, epoch_t lpr)
+    : pg(p), oid(o), last_peering_reset(lpr),
+      tid(0)
```

Comment:
Add offset as an argument and initialize here.  I see only one user which sets offset immediately after calling the constructor, so this doesn't appear to be a problem.  It's basically never ok to leave a struct/class member uninitialized after the constructor.

Reply:
You've initialized it to zero, but I want you to *pass in* the offset and initialize it *to the correct value*.  You use this in exactly one place:

```
      C_SetDedupChunks *fin = new C_SetDedupChunks(this, soid, get_last_peering_reset());
      fin->offset = p.first;
```

and then immediately reset offset to p.first.  Pass p.first into the constructor.